include/onnx-mlir/Compiler/OMCompilerTypes.h

```diff
@@ -53,6 +53,13 @@ typedef enum {
   InstrumentReportMemory
 } InstrumentActions;
 
+typedef enum {
+  afterOnnxToOnnx,
+  nnpaAfterOnnxToOnnx,
+  nnpaAfterOnnxToZhigh,
+  nnpaAfterZhighToZlow
```

Comment:
This does not conform to our policy in supporting accelerator. So `nnpa` should not appear if we do not build the compiler with `--ONNX_MLIR_ACCELERATORS=NNPA`. Everything about `nnpa` should be inside the NNPA accelerator folder and work in a plug-and-play manner.

Reply:
This is a tricky issue. NNPA has its own main and accordingly should have its own options for the main. I remember that it was discussed to separate the options for passes from those for main. Currently, there is no NPAACompilerOptions.*. However, for this case, we need to extend the enum type,  similar to the type extension in MLIR. It's too complicated for our purpose. Perhaps, we just use string type for the option to keep the choices open. It's simply but user will not know what are the right options to provide. I can not think of a better solution unless the option utility provides some support.

Reply:
Following this issue, we can add some explanation and example of instrumentStage when an accelerator is enabled or disabled (using NNPA as an example)

Reply:
`--maccel` is perhaps a good example about inserting enum values for NNPA while building the compiler: https://github.com/onnx/onnx-mlir/blob/main/src/Compiler/CompilerOptions.cpp#L98

Reply:
Since different accelerator may have different dialect and different points for instrumentation,  we need to include the real information from each accelerator directory that is in the accelerator list. Then put them together into an enum type for instrumentation stage. There may be other values needed to be put together for other purpose.  I do not know how to do it in a simple way. I am OK with that this PR may have some flaw on this option if we do not have a good solution.

Reply:
Can we use weak definition in src/Compiler and then redefine instrumentStage in NNPA/Compiler?

Reply:
Thanks for your comments!
`APPLY_TO_ACCELERATORS()` used in `--maccel` looks smart solution. I'm trying to implement using it.

> Can we use weak definition in src/Compiler and then redefine instrumentStage in NNPA/Compiler?

This is what I should do. I will try. Thanks